node/repo/interface.go

```diff
@@ -54,7 +55,7 @@ type LockedRepo interface {
 	Datastore(namespace string) (datastore.Batching, error)
```

Comment:
This looks a bit unlevelled with the Datastore method. They are both backed by some kind of store that would benefit from a cancellatio handle, so we should also propagate a context here.

Reply:
Generally fair, though not within scope of what I am trying to push out at present.
I might circle back to this at a later date when I look into migrating the datastore to be rdbms backed as wel, but I am nowhere near there yet.

@raulk do you consider lack of ctx here blocker for the PR ?

Reply:
I'm happy to push this to this PR. If we don't do it now, we are piling up one more piece of technical debt IMO.

Reply:
@raulk if you have the bandwidth - by all means, use the same PR